nifi-framework-bundle/nifi-framework/nifi-framework-components/src/main/java/org/apache/nifi/flow/synchronization/StandardVersionedComponentSynchronizer.java

```diff
@@ -1685,10 +1685,19 @@ public void synchronize(final ParameterContext parameterContext, final Versioned
 
     protected Set<String> getUpdatedParameterNames(final ParameterContext parameterContext, final VersionedParameterContext proposed) {
         final Map<String, ParameterValueAndReferences> originalValues = new HashMap<>();
-        parameterContext.getParameters().values().forEach(param -> originalValues.put(param.getDescriptor().getName(), getValueAndReferences(param)));
+        parameterContext.getEffectiveParameters().forEach((pd, param) -> originalValues.put(pd.getName(), getValueAndReferences(param)));
```

Comment:
As this code part is essentially reused what do you think about just extracting it to its own helper function like:
```
    private void collectValueAndReferences(final ParameterContext parameterContext, final Map<String, ParameterValueAndReferences> valueAndRef) {
        parameterContext.getEffectiveParameters()
                .forEach((pd, param) -> valueAndRef.put(pd.getName(), getValueAndReferences(param)));
    }
```

You could go one step further and collecting / returning the map from the function (not expecting it as parameter) but that would mean some flatMap in the logic below and I'm not sure what is the appetite for that.